test/acceptance/features/bindAppToService.feature

```diff
@@ -217,3 +217,177 @@ Feature: Bind an application to a service
+                    strategy: deployment
+                customresourcedefinitions:
+                    owned:
+                        - name: backservs.service.example.com
+                          version: v1
+                          kind: Backserv
+                          statusDescriptors:
+                            - description: Name of the Secret to hold the DB user and password
+                              displayName: DB Password Credentials
+                              path: secret
+                              x-descriptors:
+                              - urn:alm:descriptor:io.kubernetes:Secret
+                              - binding:env:object:secret:username
+                              - binding:env:object:secret:password
+                            - description: Name of the ConfigMap to hold the DB config
+                              displayName: DB Config Map
+                              path: configmap
+                              x-descriptors:
+                              - urn:alm:descriptor:io.kubernetes:ConfigMap
+                              - binding:env:object:configmap:db_host
+                              - binding:env:object:configmap:db_port
+            """
+        * The CRD "backservs.service.example.com" is present
+            """
+            apiVersion: apiextensions.k8s.io/v1beta1
+            kind: CustomResourceDefinition
+            metadata:
+                name: backservs.service.example.com
+            spec:
+                group: service.example.com
+                versions:
+                    - name: v1
+                      served: true
+                      storage: true
+                scope: Namespaced
+                names:
+                    plural: backservs
+                    singular: backserv
+                    kind: Backserv
+                    shortNames:
+                    - bs
+            """
+        * The CR "demo-backserv-cr-2" for kind "Backserv" is present
+            """
+            apiVersion: service.example.com/v1
+            kind: Backserv
+            metadata:
+                name: demo-backserv-cr-2
+            status:
+                secret: csv-demo-secret
+                configmap: csv-demo-cm
+            """
+        * The "configmap" "csv-demo-cm" is present
+            """
+            apiVersion: v1
+            kind: ConfigMap
+            metadata:
+                name: csv-demo-cm
+            data:
+                db_host: 172.72.2.0
+                db_port: "3306"
+            """
+        * The "secret" "csv-demo-secret" is present
+            """
+            apiVersion: v1
+            kind: Secret
+            metadata:
+                name: csv-demo-secret
+            type: Opaque
+            stringData:
+                username: admin
+                password: secret123
+            """
+        When Service Binding Request is applied to connect the database and the application
```

Comment:
We should have a more generic step saying that `Service Binding Request is applied`

Reply:
Yes but I think it'll be a good idea to do this in separate PR and we can also refactor some more methods there. Anyway, I was thinking of refactoring acceptance test.

Reply:
We should do it here because the used step does not reflect the reality - there is not application selector in the given yaml - so we do not conntect database and application here.

Reply:
Done.